openedx/core/lib/block_structure/transformers.py

```diff
@@ -105,3 +110,40 @@ def is_collected_outdated(cls, block_structure):
             )
 
         return bool(outdated_transformers)
+
+    def _transform_optimized(self, transformers, block_structure):
+        """
+        Transforms the given block_structure using the transform_block_filter
+        method from the given transformers.
+        """
+        if not transformers:
+            return
+
+        removal_filters = []
+        for removal in [t.transform_block_filter(self.usage_info, block_structure) for t in transformers]:
+            if isinstance(removal, list):
+                removal_filters.extend(removal)
+            else:
+                removal_filters.append(removal)
+
+        combined_filters = reduce(
+            lambda t1_filter, t2_filter: self._filter_chain(t1_filter, t2_filter),
```

Comment:
Honestly, I'd try taking at least one stab at writing it out as a plain for loop. Half the people looking at this code later are going to have to lookup the docs for `functools.reduce` because they never use it in their day to day coding.